examples/chef/common/clusters/oven-cavity-operational-state/chef-oven-cavity-operational-state.cpp

```
@@ -0,0 +1,317 @@
+ *    limitations under the License.
+ */
+
+#include "chef-oven-cavity-operational-state.h"
+#include <app/util/attribute-storage.h>
+#include <app/util/endpoint-config-api.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#ifdef MATTER_DM_PLUGIN_OVEN_CAVITY_OPERATIONAL_STATE_SERVER
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OvenCavityOperationalState {
+
+constexpr size_t kOvenCavityOperationalStateTableSize = MATTER_DM_OPERATIONAL_STATE_OVEN_CLUSTER_SERVER_ENDPOINT_COUNT;
+static_assert(kOvenCavityOperationalStateTableSize <= kEmberInvalidEndpointIndex, "OvenCavityOperationalState table size error");
+
+std::unique_ptr<ChefDelegate> gDelegateTable[kOvenCavityOperationalStateTableSize];
+std::unique_ptr<Instance> gInstanceTable[kOvenCavityOperationalStateTableSize];
+
+static void onOvenCavityOperationalStateTimerTick(System::Layer * systemLayer, void * data)
+{
+    ChefDelegate * delegate = reinterpret_cast<ChefDelegate *>(data);
+
+    uint8_t opState = delegate->GetCurrentOperationalState();
+    if (opState != to_underlying(OperationalStateEnum::kRunning))
+    {
+        ChipLogError(DeviceLayer, "onOperationalStateTimerTick: Operational cycle can not be active in state %d", opState);
+        (void) DeviceLayer::SystemLayer().CancelTimer(onOvenCavityOperationalStateTimerTick, delegate);
+        return;
+    }
+
+    if (!delegate->CheckCycleActive())
+    {
+        (void) DeviceLayer::SystemLayer().CancelTimer(onOvenCavityOperationalStateTimerTick, delegate);
+        return;
+    }
+
+    delegate->CycleSecondTick();
+
+    if (delegate->CheckCycleComplete())
+    {
+        OperationalState::GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+        delegate->HandleStopStateCallback(err);
+    }
+    else
+    {
+        (void) DeviceLayer::SystemLayer().StartTimer(System::Clock::Seconds16(1), onOvenCavityOperationalStateTimerTick, delegate);
```

What is the lifetime/what controls the timer? The code above seems to assume the timer is always active (we issue CancelTimer) where as this seems to assume the timer is not active and resets it to 1 second.

Some comments explaining at the top of the method might help: who starts it, is it one-shot or continous, who is responsible to stop it or restart it. It seems we want a tick every second.

---

Added comments.

---

Specifically I am trying to understand `why are we calling CancelTimer above`?

https://github.com/project-chip/connectedhomeip/blob/master/src/system/SystemLayer.h#L119 says `This method starts a one-shot timer`. So having StartTimer makes sense.

Separately? Why are we silently ignoring errors on things? Could we at least `LogErrorOnFailure` instead of `(void)` on things? Does that increase flash too much?